sigar_interface.go

```diff
@@ -10,6 +10,15 @@ type Sigar interface {
 	GetMem() (Mem, error)
 	GetSwap() (Swap, error)
 	GetFileSystemUsage(string) (FileSystemUsage, error)
+	GetFDUsage() (FDUsage, error)
+}
+
+type ErrNotImplemented struct {
+	msg string
```

Comment:
Consider making this parameter be the OS. Then make the `Error()` function `return "not implemented on " + e.OS`. There's a bit of duplication in the error messages at the moment.